earn/src/components/portfolio/modal/EarnInterestModal.tsx

```diff
@@ -285,8 +307,8 @@ export default function EarnInterestModal(props: EarnInterestModalProps) {
             <BaseMaxButton
               size='L'
               onClick={() => {
-                if (depositBalance != null) {
-                  setInputValue(depositBalance?.formatted);
+                if (tokenBalanceResult != null) {
+                  setInputValue(userBalance.toString(GNFormat.DECIMAL));
```

Comment:
While I see why it was done this way, I wish we could avoid having seemingly unrelated variables used in the if statement (despite them being quite related in actuality).

Reply:
Hmm good point. This was actually just a result of refactoring; I didn't write the code, but now that you point it out it is kinda bad/confusing. Any suggestions to fix?